### PR TITLE
Eu opinions download

### PR DESCRIPTION
**Pull request by Levia**
## Description

First draft of the backend work necessary to include SRG History field into the CSV downloads as well as considering other possible filtering options for the Eu Decision Type (e.g. the new SRG Referral).
'Discussed at SRG' EU decisions are now also excluded.

### CODE REVIEW

**Comment by lucacug** [comment]
@Levia  Made a few tests and looks good to me. Could you please have a look when you get the chance ?

File: app/assets/stylesheets/species/downloads.scss [hunk 1]
```diff
@@ -379,3 +379,13 @@
 		}
 	}
 }
+
+.download-block {
+	.sub-heading {
+		color: #2d3237;
+		font-weight: bold;
+		text-transform: uppercase;
+		user-select: none;
```
Comment (Levia) [inline app/assets/stylesheets/species/downloads.scss:388]: I feel there's too much indentation here perhaps?

File: app/models/species/eu_decisions_export.rb [hunk 2]
```diff
@@ -6,15 +6,18 @@ def initialize(filters)
     @geo_entities_ids = filters[:geo_entities_ids]
     @years = filters[:years]
     @decision_types = filters[:decision_types]
+    @in_consultation = filters[:eu_decision_filter]
     @set = filters[:set]
     initialize_csv_separator(filters[:csv_separator])
     initialize_file_name
   end
 
+  HISTORIC_III_OPINIONS = ['(No opinion) iii)','(No opinion) iii) removed'].freeze
```
Comment (Levia) [inline app/models/species/eu_decisions_export.rb:15]: I think the only doubt I have is related to the `iii) removed` which I'm not sure are intended to be included as well when fetching the SRG Referral data. Unless this has already been checked with the client, which would then be fine.

File: app/models/species/eu_decisions_export.rb [hunk 3]
```diff
@@ -6,15 +6,18 @@ def initialize(filters)
...
   def query
     rel = EuDecision.from("#{table_name} AS eu_decisions").
       select(sql_columns).
       order(:taxonomic_position, :party, :ordering_date)
+    return rel.where('srg_history =?', @in_consultation) if @in_consultation == 'In consultation'
```
Comment (Levia) [inline app/models/species/eu_decisions_export.rb:20]: Perhaps the variable name might be a bit confusing for the way it is used.
I would either call it like the param, so `eu_decision_filter`, in a way that is clearer that the value contains one of the possible options for that, or make this a boolean somehow. 
The former might be easier for how it is currently implemented.

Also, to be extremely picky, I would add a space between `=` and `?` :)

**Comment by lucacug** [comment]
@Levia I see what you mean. When selecting historic no opinions and current at the same time, you will still get all the No opinion i) that are not really historical. I would suggest to leave it as it is for the moment and trying to deploy to staging, if not today Monday morning, so that they can check by themself. What do you think?

File: app/models/species/eu_decisions_export.rb (hunk 2 above)
Comment (lucacug) [reply app/models/species/eu_decisions_export.rb:15]: As the iii) removed are the actual historic values at the moment, they appear in the download for the No opinions All filter. With the introduction of the SRG Referral also the actual iii) opinions will become historical, so my doubt would be that the difference between iii) and iii) removed will not make much sense anymore as will both be historical.

**Comment by Levia** [comment]
@lucacug Thanks for claryfing! Yeah I agree, let's deploy this to staging asap so they can check.
Happy for you to merge and deploy when more convenient!

**Pull request merged**